### ISSUE TITLE
"no-unused-vars" is coming up with a false positive

### ISSUE THREAD

**Issue by christo8989** [eslint/eslint#13916]
Environment Info:

```
Node version: v11.15.0
npm version: v6.7.0
Local ESLint version: v7.15.0 (Currently used)
```

Parser
`"parser": "babel-eslint",`

<details>
<summary>Configuration</summary>

```js
{
  "env": {
    "browser": true,
    "es2021": true,
    "jest": true
  },
  "extends": [
    "react-app",
    "plugin:prettier/recommended"
  ],
  "plugins": [
    "prettier"
  ],
  "parser": "babel-eslint",
  "rules": {
    "no-debugger": "error",
    "prettier/prettier": ["error"],
    "no-anonymous-default-export": 0,
    "import/no-anonymous-default-export": 0,
    "react-hooks/exhaustive-deps": 0,
    "react-hooks/rules-of-hooks": 0
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
  if (graphQLErrors) {
    for (let err of graphQLErrors) {    // errors on this line with no-unused-vars
      if (err && err.extensions) {
        switch (err.extensions.code) {
          // code
        }
      }
    }
  }
```

```bash
eslint --fix --max-warnings 0 src
```

**What did you expect to happen?**
Because the property is being used, I expected no errors or warnings.

**What actually happened? Please include the actual, raw output from ESLint.**
It gives a warning that `err` is an unused variable.

**Are you willing to submit a pull request to fix this bug?**
Not at the moment.

**Comment by anikethsaha** [comment]
Thanks for the issue. 

This seems like an issue with `babel-eslint` as this is working fine with eslint's default parser 

[demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLXZhcnM6IFwiZXJyb3JcIiovXG5cbiBpZiAoZ3JhcGhRTEVycm9ycykge1xuICAgIGZvciAobGV0IGVyciBvZiBncmFwaFFMRXJyb3JzKSB7ICBcbiAgICAgIGlmIChlcnIgJiYgZXJyLmV4dGVuc2lvbnMpIHtcbiAgICAgICAgc3dpdGNoIChlcnIuZXh0ZW5zaW9ucy5jb2RlKSB7XG4gICAgICAgICAgLy8gY29kZVxuICAgICAgICB9XG4gICAgICB9XG4gICAgfVxuICB9Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Comment by mdjermanovic** [comment]
Hi @christo8989!

You are probably using an older version of `babel-eslint`, this problem was fixed in v10.0.3. Can you try with the latest version? 

Also, `babel-eslint` is now [@babel/eslint-parser](https://www.npmjs.com/package/@babel/eslint-parser).

**Comment by christo8989** [comment]
Updating packages fixed the issue for me.
Thank you for the help!
